editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,450 @@
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "group_settings_editor.h"
+
+#include "core/config/project_settings.h"
+#include "editor/editor_scale.h"
+#include "editor/editor_undo_redo_manager.h"
+#include "editor/scene_tree_dock.h"
+#include "editor_file_system.h"
+#include "editor_node.h"
+#include "scene/resources/packed_scene.h"
+
+void GroupSettingsEditor::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE: {
+			update_groups();
+		} break;
+	}
+}
+
+void GroupSettingsEditor::_item_edited() {
+	if (updating_group) {
+		return;
+	}
+
+	TreeItem *ti = tree->get_edited();
+	int column = tree->get_edited_column();
+
+	if (!ti) {
+		return;
+	}
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	if (column == 0) {
+		// Name Edited.
+		String new_name = ti->get_text(0);
+		String old_name = ti->get_meta("__name");
+
+		if (new_name == old_name) {
+			return;
+		}
+
+		String error = _check_new_group_name(new_name);
+		if (!error.is_empty()) {
+			ti->set_text(0, old_name);
+			show_message(error);
+			return;
+		}
+
+		String old_property_name = GLOBAL_GROUP_PREFIX + old_name;
+		String new_property_name = GLOBAL_GROUP_PREFIX + new_name;
+
+		String description = ti->get_meta("__description");
+
+		undo_redo->create_action(TTR("Rename Group"));
+
+		undo_redo->add_do_property(ProjectSettings::get_singleton(), new_property_name, description);
+		undo_redo->add_undo_property(ProjectSettings::get_singleton(), new_property_name, Variant());
+
+		undo_redo->add_do_property(ProjectSettings::get_singleton(), old_property_name, Variant());
+		undo_redo->add_undo_property(ProjectSettings::get_singleton(), old_property_name, description);
+
+		undo_redo->add_do_method(this, "rename_references", old_name, new_name);
+		undo_redo->add_undo_method(this, "rename_references", new_name, old_name);
```

Comment:
A method that scans every scene is part of undoredo action 😬
It's nice that you can undo it, but going through all scenes can take many minutes in a big project. I don't know how to handle it better though, other than making a separate refactoring tool. I guess it's something for another PR.